src/main/python/camdkit/framework.py

```diff
@@ -233,30 +312,338 @@ def make_json_schema() -> dict:
       "additionalProperties": False
     }
 
-class StrictlyPositiveIntegerParameter(Parameter):
+class IntegerParameter(Parameter):
+  """Base class for integer parameters"""
+
+  @staticmethod
+  def to_json(value: typing.Any) -> typing.Any:
+    return value
+
+  @staticmethod
+  def from_json(value: typing.Any) -> typing.Any:
+    return int(value)
+  
+class NonNegativeIntegerParameter(IntegerParameter):
 
   @staticmethod
   def validate(value) -> bool:
     """The parameter shall be a integer in the range (0..2,147,483,647]."""
 
+    return isinstance(value, numbers.Integral) and value >= 0
+
+  @staticmethod
+  def make_json_schema() -> dict:
+    return {
+      "type": "integer",
+      "minimum": 0,
+      "maximum": INT_MAX
+    }
+  
+class StrictlyPositiveIntegerParameter(IntegerParameter):
+
+  @staticmethod
+  def validate(value) -> bool:
+    """The parameter shall be a integer in the range (1..2,147,483,647]."""
+
     return isinstance(value, numbers.Integral) and value > 0
 
+  @staticmethod
+  def make_json_schema() -> dict:
+    return {
+      "type": "integer",
+      "minimum": 1,
+      "maximum": INT_MAX
+    }
+  
+class NonNegativeRealParameter(Parameter):
+  
+  @staticmethod
+  def validate(value) -> bool:
+    """The parameter shall be a non-negative real number."""
+
+    return isinstance(value, numbers.Real) and value >= 0.0
+
   @staticmethod
   def to_json(value: typing.Any) -> typing.Any:
     return value
 
   @staticmethod
   def from_json(value: typing.Any) -> typing.Any:
-    return int(value)
+    return float(value)
+  
+  @staticmethod
+  def make_json_schema() -> dict:
+    return {
+      "type": "number",
+      "minimum": 0.0,
+    }
+  
+class EnumParameter(StringParameter):
+  allowedValues = []
+  enum_class = None
+
+  @classmethod
+  def __init_subclass__(cls) -> None:
+    if cls.enum_class == None:
+      # Sub-class must specify the related Enum class
+      raise ValueError
+    cls.allowedValues = [e.value for e in cls.enum_class]
+
+  def validate(self, value) -> bool:
+    """The parameter shall be one of the allowed values."""
+    return str(value) in self.allowedValues
+
+  def make_json_schema(self) -> dict:
+    return {
+      "type": "string",
+      "enum": self.allowedValues
+    }
+  
+class TimingModeParameter(EnumParameter):
```

Comment:
No need to define this class when there is a single derived class. Also, the sampling should be specified in the model so that the same type can be used for both dynamic and static parameters.

Reply:
I've moved them to the model if there is only a single derived class